.github/workflows/buildtunebook.yaml

```diff
@@ -0,0 +1,65 @@
+name: Build Tunebook
+
+on:
+  pull_request:
+  workflow_dispatch:
+  push:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    name: Setup Environment
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: '16'
+
+      - name: checks
+        run: |
+          echo "Hello"
+          tree
+
+      - name: cache npm
+        uses: actions/cache@v2
+        with:
+          path: '**/node_modules'
+          key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
+
+      - name: pre install
+        run: |
+          sudo apt install abcm2ps
```

Comment:
@Nick-Pearson In theory using sudo creates a vulnerability to change, however I think these very old scripts are being maintained well - they seem the basis of far too many other ABC programs. I've left the sub-modules for safety's sake, lest we need to go back to the old method.

Reply:
Yes seems fair enough to me